core/src/main/java/org/mule/api/el/ExpressionLanguage.java

```diff
@@ -109,4 +110,5 @@
     @Deprecated
     <T> T evaluate(String expression, MuleMessage message, Map<String, Object> vars);
 
+    TypedValue evaluateTyped(String expression, MuleMessage message);
```

Comment:
I think this should receive the event, not the message. I know that might lead to inconsistency with other similar classes and that it might even be impossible due to the contract that collaborators of this class might have. But I'd like to raise that point anyhow because MEL currently has issues because of this duality. I've seen some functions failed because the _muleEvent context variable not always exists.